### DISCUSSION TITLE
Web Mode: Could Mu include a default app.py?

### DISCUSSION THREAD

**Discussion by prcutler** [mu-editor/mu#2320] [Ideas & New Features]
I'm playing around with Web mode as it needs a tutorial written (issue #2271 ) and Mu ships with HTML files and directories (static, templates) with a "Hello from Mu" page and sub-pages.  If a basic app.py was included, when the user hits "Run" in Mu, it would load the "Hello from Mu" page.  The code that worked for me looks like:

```
# A simple web application.
from flask import Flask, render_template

app = Flask(__name__)

@app.route("/")
def index():
    return render_template('index.html')
```

Is there interest in possibly including something like this to help get users one step closer to running a Flask app?

**Comment by carlosperate** [comment]
I believe if you are in the "Web mode" and click the "New" button it will open a new tab with this Python code by default, no?

```python
# A simple web application.
from flask import Flask, render_template


app = Flask(__name__)


@app.route("/")
def index():
    return render_template('index.html')

```

Or maybe I misunderstood what you meant?

**Comment by prcutler** [comment]
You're correct, I think I didn't understand.

When I click new, I do get the code you mention.  But when I hit "Run", it opens up a save dialog, which is what led to my confusion.  And I think it would lead to new user confusion - what are they supposed to be saving? (File name, etc)

I hope that makes sense.

**Reply by carlosperate** [reply]
Right, I see what you mean.

I think this has been a patterns Mu had for a while, the "new" button can create a new file with template code, that the user still needs to save it to be able to run it.

Creating a new file by default might bring other issues, like what happens if there is alredy an `app.py` file in the `mu_code` folder?

The current approach is probably the best compromise we've come up with so far, but happy to revisit it if there any good alternatives!
